apps/studio/components/interfaces/Auth/Policies/Policies.utils.ts

```diff
@@ -151,3 +155,277 @@ export const createPayloadForUpdatePolicy = (
 
   return payload
 }
+
+// --- Policy Generation ---
+
+/**
+ * Generated policy extends CreatePolicyBody with additional fields for display.
+ * - sql: Full CREATE POLICY SQL statement for preview
+ * - Required fields that are optional in CreatePolicyBody
+ */
+export type GeneratedPolicy = Required<
+  Pick<CreatePolicyBody, 'name' | 'table' | 'schema' | 'action' | 'roles'>
+> &
+  Pick<CreatePolicyBody, 'command' | 'definition' | 'check'> & {
+    sql: string
+  }
+
+type Relationship = {
+  source_schema: string
+  source_table_name: string
+  source_column_name: string
+  target_table_schema: string
+  target_table_name: string
+  target_column_name: string
+}
+
+/**
+ * Gets relationships for a specific table from FK constraints.
+ * Returns relationships where the table is the source.
+ */
+const getRelationshipsForTable = (
+  schema: string,
+  tableName: string,
+  allForeignKeyConstraints: ForeignKeyConstraint[]
+): Relationship[] => {
+  return allForeignKeyConstraints
+    .filter((fk) => fk.source_schema === schema && fk.source_table === tableName)
+    .flatMap((fk) =>
+      fk.source_columns.map((sourceCol, i) => ({
+        source_schema: fk.source_schema,
+        source_table_name: fk.source_table,
+        source_column_name: sourceCol,
+        target_table_schema: fk.target_schema,
+        target_table_name: fk.target_table,
+        target_column_name: fk.target_columns[i],
+      }))
+    )
+}
+
+/**
+ * BFS to find shortest path from table to auth.users via foreign key relationships.
+ * Returns null if no path exists within maxDepth.
+ */
+const findPathToAuthUsers = (
```

Comment:
this particular method feels overly complicated at first glance and i can't really tell what's it meant to do

what does `path` mean here?
`const path = findPathToAuthUsers(table, foreignKeyConstraints)`